dashboard/apps/dashboard/management/commands/sync.py

```diff
@@ -111,6 +112,28 @@ def sync_clients(data_dir):
             client.save()
 
 
+def sync_departments(data_dir):
+    logging.info('sync departments')
+    source = os.path.join(data_dir, 'departments.json')
+    with open(source, 'r') as sf:
+        data = json.loads(sf.read())
+    for item in data:
+        useful_data = {
+            'float_id': item['department_id'],
+            'name': item['department_name'],
+            'raw_data': item,
+        }
+        try:
+            department = Department.objects.get(
+                float_id=useful_data['float_id'])
+            diff = compare(department, useful_data)
```

Comment:
Looks like the `compare` function has side effects.

It updates the `existing_object` (`department` in this case) but does not commit the changes. Then the `update` function, logs and commits the changes.

I think the code that updates the existing object (https://github.com/ministryofjustice/moj-product-dashboard/dashboard/apps/dashboard/management/commands/sync.py#L58) should be moved to `update`